prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+    generate_dhw_profiles,
+)
+from prereise.gather.demanddata.bldg_electrification.ff2elec_profile_generator_htg import (
+    generate_htg_profiles,
+)
+from prereise.gather.demanddata.bldg_electrification.helper import (
+    read_shapefile,
+    zone_shp_overlay,
+)
+from prereise.gather.demanddata.bldg_electrification.zone_profile_generator import (
+    zonal_data,
+)
+
+
+class Scenarios:
+    # read and process a scenario (reflects either model base year or a projection year) from user defined parameters
+    def __init__(self, id, input_series, other=None):
+        """define scenarios.
+            For a base scenario, read and save building stock inputs
+            For a projection scenario, some fields are projected from base scenario based on the projection scenario inputs
+
+        #param str id: id of scenario, 'base' refers to modeled base year scenario
+        #param Pandas Series input_series: contains information of building stock energy usages from scenario inputs
+        #param class scenarios other: base scenario. When other == None, then define a base scenario from scenario inputs
+        """
+        self.id = id
+        self.year = int(input_series["year"])
+        self.hp_type_heat = input_series.pop("heat_hp_type")
+        self.hp_type_dhw = input_series.pop("dhw_hp_type")
+        self.cook_efficiency = input_series.pop("cook_eff")
+        input_series = input_series.astype("float64")
+        self.cool_energy_intensity = input_series["cool_energy_intensity(relative)"]
+        self.stats = input_series
+        if other is None:
+            self.stats = (
+                input_series.dropna()
+            )  # dropna is supposed to drop rows that only used for defining future scenarios
+            self._compute_base_scenario()
+
+        else:
+            self._compute_project_scenario(other)
+
+    def _compute_base_scenario(self):
+        self.pop = self.stats["pop"]
+        self.floor_area_m2 = {
+            "res": self.stats["res_area_m2"],
+            "com": self.stats["com_area_m2"],
+        }
+        self.hp_heat_area_m2 = (
+            self.stats["frac_hp_res_heat"] * self.floor_area_m2["res"]
+            + self.stats["frac_hp_com_heat"] * self.floor_area_m2["com"]
+        )
+        self.hp_heat_frac = self.hp_heat_area_m2 / (
+            self.floor_area_m2["res"] + self.floor_area_m2["com"]
+        )
+        self.resist_heat_area_m2 = (
+            self.stats["frac_resist_res_heat"] * self.floor_area_m2["res"]
+            + self.stats["frac_resist_com_heat"] * self.floor_area_m2["com"]
+        )
+        self.resist_heat_frac = self.resist_heat_area_m2 / (
+            self.floor_area_m2["res"] + self.floor_area_m2["com"]
+        )
+        self.elec_cool_m2 = (
+            self.stats["frac_elec_res_cool"] * self.floor_area_m2["res"]
+            + self.stats["frac_elec_com_cool"] * self.floor_area_m2["com"]
+        )
+        return
```

Comment:
I think we don't need this `return` here.

Reply:
done